js/script.js

```diff
@@ -0,0 +1,481 @@
+            );
+                             
+            break;
+
+          case 'remove':             
+            if(isNaN(key)){ 
+              matchText = JSON.stringify(key) + ': ' +JSON.stringify(this.jsonPatch[index].previousValue); 
+            }
+            else{ 
+              matchText = JSON.stringify(this.jsonPatch[index].previousValue); 
+            }                    
+            
+            this.updateHTMLOutput(this.jsonObj); // Temporarily updating the DOM with formatted Object before change to display the change.
+            this.jsonHTML = this.jsonHTML.replace( // Displaying the change using CSS
+              matchText,'<span class="old">' + matchText + '</span>'
+            );         
+            break;
+          case 'replace':
+            this.jsonHTML = this.jsonHTML.replace( // Displaying the change using CSS showing old and updated Value.
+              JSON.stringify(value),             
+                (this.jsonPatch[index].previousValue !== undefined ? '<span class="old">' + this.jsonPatch[index].previousValue + '</span> ' : '')
+              + '<span class="new">' + value + '</span>'
+            );        
+            break;
+          case 'move':
+            fromKey = fromPath[fromPath.length - 1];  
+            let tempObj = this.copy(this.jsonObj);
+
+            tempObj = this.setValue(path,this.getValue(fromPath,tempObj),tempObj);      
+            this.updateHTMLOutput(tempObj);                  
+
+            if(isNaN(key)){
+              matchText = JSON.stringify(fromKey) + ': ' +JSON.stringify(this.getValue(fromPath,this.jsonObj)); 
+            }
+            else{ 
+              matchText = JSON.stringify(this.getValue(fromPath,this.jsonObj)); 
+            }
+
+            console.log(matchText);
+
+            this.jsonHTML = this.jsonHTML.replace(
+              matchText,'<span class="old">' + matchText + '</span>'
+            );
+            break;
+          case 'copy':      
+            fromKey = fromPath[fromPath.length - 1];   
+            if(isNaN(fromKey)){ 
+              matchText = JSON.stringify(fromKey) + ': ' +JSON.stringify(this.getValue(fromPath,this.newJsonObj)); 
+            }
+            else{ 
+              matchText = JSON.stringify(this.getValue(fromPath,this.newJsonObj)); 
+            }          
+
+            this.updateHTMLOutput(this.newJsonObj);                   
+            this.jsonHTML = this.jsonHTML.replace(
+              matchText,'<span class="new">' + matchText + '</span>'
+            );
+            break;
+          case 'test':
+            break;
+        }
+      }
+      else{
+        alert('You should execute the patches in order');
+      }
+    },
+
+    getPath(path){ // Get the JSON patch string from the array of Path. eg [a,b,c,1] should return a.b.c[1];
+      let pathString = path[0];
+      for(let i=1;i< path.length;i++){
+        if(isNaN(path[i])){
+          pathString += '.' + path[i];
```

Comment:
a property name that has space or a hyphen cannot be accessed using dot notation

Reply:
Understood.